spec/hash-map.dd

```diff
@@ -337,8 +337,6 @@ $(H2 $(LNAME2 advanced_updating, Advanced updating))
 
 $(H2 $(LNAME2 static_initialization, Static Initialization of AAs))
 
-        NOTE: Not yet implemented.
```

Comment:
Firstly, "not yet implemented" is terrible for a language specification - so getting rid of it is good. 

This behaviour needs to be better described - also a note on the word behaviour: In the context of a language *specification* the aim is not to describe what the language does in reality, but what it should do in order for it to be consistent in some desirable way.

https://dlang.org/spec/declaration.html#global_static_init The spec declares that this behaviour should be implemented for D to be compliant with its own specification in the sense that this declaration should be evaluable at compile time - in this case the expression is evaluable but for whatever (druntime) reason cannot be assigned to anything global or static.

The behaviour needs to be elaborated upon and the non-compliance noted in a more concrete way with an example.

Reply:
I elaborated a bit more, similarly to static initialization of normal arrays. I also added a more elaborated note about not being implemented on dmd compiler.